pkg/api/grpc/proxy/handler_test.go

```diff
@@ -833,7 +832,7 @@ func (s *proxyTestSuite) SetupSuite() {
 
 	clientConn, err := grpc.DialContext(
 		context.Background(),
-		strings.Replace(s.proxyListener.Addr().String(), "127.0.0.1", "localhost", 1),
+		s.proxyListener.Addr().String(), // DO NOT USE "localhost" as it does not resolve to loopback in some environments.
```

Comment:
I don't think this will work on IPv6-only networks?

Reply:
I remember these fixing flakiness here. I will run another PR without the localhost changes for us to compare now that we have a green baseline. If ipv6 only networks is the concern (although rare for local networks), I can take this value from an env variable.

Reply:
Let's compare: https://github.com/dapr/dapr/pull/7492

Reply:
As I mentioned in Discord, this is only for tests and not in runtime code.